docs/template_plugin/tests/functional/op_reference/acosh.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <vector>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+
+namespace {
+
+template <typename T>
+std::vector<T> create_iota_vector(size_t size, T first_value = {}) {
+    std::vector<T> d(size);
+    std::iota(begin(d), end(d), first_value);
+    return d;
+}
+
+template <typename T>
+struct Iota {
+    Iota(T e = {}): first_element_value(e) {}
+    T first_element_value;
+};
```

Comment:
I'm not sure you really need this `Iota` struct

Reply:
It's not, will be remove.